File: sdks/java/lib/src/main/java/models/shared/ShieldInformationBarrierReport.java
package .models.shared;

import java.time.OffsetDateTime;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import .utils.DateTimeSerializer;
import .utils.DateTimeDeserializer;

/**
 * ShieldInformationBarrierReport
 * A base representation of a
 * shield information barrier report object
**/
public class ShieldInformationBarrierReport {
    @JsonInclude(Include.NON_ABSENT)
    @JsonSerialize(contentUsing = DateTimeSerializer.class)
    @JsonDeserialize(contentUsing = DateTimeDeserializer.class)
    @JsonProperty("created_at")
    public OffsetDateTime createdAt;
    public ShieldInformationBarrierReport withCreatedAt(OffsetDateTime createdAt) {
        this.createdAt = createdAt;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("created_by")
    public UserBase createdBy;
    public ShieldInformationBarrierReport withCreatedBy(UserBase createdBy) {
        this.createdBy = createdBy;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("details")
    public ShieldInformationBarrierReportDetails details;
    public ShieldInformationBarrierReport withDetails(ShieldInformationBarrierReportDetails details) {
        this.details = details;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("id")
    public String id;
    public ShieldInformationBarrierReport withId(String id) {
        this.id = id;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("shield_information_barrier")
    public ShieldInformationBarrierReference shieldInformationBarrier;
    public ShieldInformationBarrierReport withShieldInformationBarrier(ShieldInformationBarrierReference shieldInformationBarrier) {
        this.shieldInformationBarrier = shieldInformationBarrier;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("status")
    public ShieldInformationBarrierReportStatusEnum status;
    public ShieldInformationBarrierReport withStatus(ShieldInformationBarrierReportStatusEnum status) {
        this.status = status;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("type")
    public ShieldInformationBarrierReportTypeEnum type;
    public ShieldInformationBarrierReport withType(ShieldInformationBarrierReportTypeEnum type) {
        this.type = type;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonSerialize(contentUsing = DateTimeSerializer.class)
    @JsonDeserialize(contentUsing = DateTimeDeserializer.class)
    @JsonProperty("updated_at")
    public OffsetDateTime updatedAt;
    public ShieldInformationBarrierReport withUpdatedAt(OffsetDateTime updatedAt) {
        this.updatedAt = updatedAt;
        return this;
    }
}

File: sdks/go/pkg/models/shared/shieldinformationbarrierreport.go
package shared

import (
	"time"
)

type ShieldInformationBarrierReportStatusEnum string

const (
	ShieldInformationBarrierReportStatusEnumPending   ShieldInformationBarrierReportStatusEnum = "pending"
	ShieldInformationBarrierReportStatusEnumError     ShieldInformationBarrierReportStatusEnum = "error"
	ShieldInformationBarrierReportStatusEnumDone      ShieldInformationBarrierReportStatusEnum = "done"
	ShieldInformationBarrierReportStatusEnumCancelled ShieldInformationBarrierReportStatusEnum = "cancelled"
)

type ShieldInformationBarrierReportTypeEnum string

const (
	ShieldInformationBarrierReportTypeEnumShieldInformationBarrierReport ShieldInformationBarrierReportTypeEnum = "shield_information_barrier_report"
)

// ShieldInformationBarrierReport
// A base representation of a
// shield information barrier report object
type ShieldInformationBarrierReport struct {
	CreatedAt                *time.Time                                `json:"created_at,omitempty"`
	CreatedBy                *UserBase                                 `json:"created_by,omitempty"`
	Details                  *ShieldInformationBarrierReportDetails    `json:"details,omitempty"`
	ID                       *string                                   `json:"id,omitempty"`
	ShieldInformationBarrier *ShieldInformationBarrierReference        `json:"shield_information_barrier,omitempty"`
	Status                   *ShieldInformationBarrierReportStatusEnum `json:"status,omitempty"`
	Type                     *ShieldInformationBarrierReportTypeEnum   `json:"type,omitempty"`
	UpdatedAt                *time.Time                                `json:"updated_at,omitempty"`
}
